lib/_stream_readable.js

```diff
@@ -170,6 +170,9 @@ function ReadableState(options, stream, isDuplex) {
   // If true, a maybeReadMore has been scheduled.
   this.readingMore = false;
 
+  // Flags to update highwatermark
+  this.updateHighwaterMark = null;
```

Comment:
both test that I added.  
There is an alternative way to solve this issue as well: if setter is called while reading is set to true we could throw an error. 

That Flag/value is there to "schedule" an update if the setter was called while we are in reading mode. so that we could update  HWM at the start of next read.

Reply:
I think Matteo meant what if we only add this property if it is set - so streams where it is called on become are a little slower (due to shapes) but we don't have any impact on anyone else?

Reply:
I might be happier with throwing as there will be less magic involved.